ddprof-exporter/src/containerid.rs

```diff
@@ -0,0 +1,125 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+use regex::Regex;
+use std::error;
+use std::fmt;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+
+// Extract container id from /proc/self/group
+// Source: https://github.com/DataDog/dd-trace-go/blob/v1/internal/container.go
+
+const CGROUP_PATH: &str = "/proc/self/cgroup";
```

Comment:
Indeed, this can be problematic for whole host profiling.

Reply:
Yes, but those scenarios aren't really common any more and our average customer uses a commoditized cloud-based VM with a commoditized cgroup management system (containers), so I wouldn't bother splitting hairs over my dumb edge cases.

Mostly, I was thinking about the exact scenario you mentioned.  Reading through your implementation more carefully after my comment, it's clear to me that the current API will work without any modifications.  I think what we'd do for the whole-host profiler is _something_ like:
* scrape the container ID for a specific PID
* collect PIDs into disjoint sets according to their container membership
* aggregate call stack information in a pprof for each container
* send each container's export separately, with its own tags
* Anything in a pure-kernel thread, or which cannot otherwise be checked for cgroup membership, can just get associated to an export which will get tagged with the profiler's own container ID

Anyway, nothing to be done about it at this point, we don't even have an end-to-end complete whole-host profiler close to shipping soon.  :)